src/hexkit/providers/mongodb/provider.py

```diff
@@ -59,8 +61,8 @@ def __init__(
         *,
         dto_model: type[Dto],
         id_field: str,
-        collection: AsyncIOMotorCollection,
-        session: Optional[AsyncIOMotorClientSession] = None,
+        collection: AsyncIOMotorCollection,  # type: ignore
+        session: Optional[AsyncIOMotorClientSession] = None,  # type: ignore
```

Comment:
To avoid the error, you can try this:

```python
from motor.core import AgnosticCollection, AgnosticClientSession
```

And then:

```python
        collection: AgnosticCollection,
        session: Optional[AgnosticClientSession] = None,
```